sc/src/test/java/uk/ac/ebi/atlas/download/FileDownloadControllerWIT.java

```diff
@@ -148,22 +154,24 @@ void downloadArchiveForInvalidExperimentAccessions() throws Exception {
                 .param("accession", INVALID_EXPERIMENT_ACCESSION_LIST.get(0))
                 .param("accession", INVALID_EXPERIMENT_ACCESSION_LIST.get(1)))
                 .andExpect(status().isOk())
-                .andExpect(header().doesNotExist("Content-Length"));
+                .andExpect(content().string(""));
     }
 
     @Test
     void downloadArchiveForMixedValidAndInvalidExperimentAccessions() throws Exception {
+        ResultActions result  = this.mockMvc.perform(get(ARCHIVE_DOWNLOAD_LIST_URL)
+                .param("accession", EXPERIMENT_ACCESSION_LIST.get(0))
+                .param("accession", EXPERIMENT_ACCESSION_LIST.get(1))
+                .param("accession", INVALID_EXPERIMENT_ACCESSION_LIST.get(0))
+                .param("accession", INVALID_EXPERIMENT_ACCESSION_LIST.get(1)));
+        byte[] contentByte = result.andReturn().getResponse().getContentAsByteArray();
+        ZipInputStream zipInputStream = new ZipInputStream(new ByteArrayInputStream(contentByte));
         String expectedArchiveName =
                 MessageFormat.format(
                         ARCHIVE_NAME, EXPERIMENT_ACCESSION_LIST.size(), "experiment");
 
-        this.mockMvc.perform(get(ARCHIVE_DOWNLOAD_LIST_URL)
-                .param("accession", EXPERIMENT_ACCESSION_LIST.get(0))
-                .param("accession", EXPERIMENT_ACCESSION_LIST.get(1))
-                .param("accession", INVALID_EXPERIMENT_ACCESSION_LIST.get(0))
-                .param("accession", INVALID_EXPERIMENT_ACCESSION_LIST.get(1)))
-                .andExpect(status().isOk())
-                .andExpect(header().string("Content-Length", String.valueOf(EXPERIMENT_ACCESSION_LIST.size() * 10)))
+        result.andExpect(status().isOk())
+                .andExpect(content().string(containsString(zipInputStream.getNextEntry().getName())))
```

Comment:
You should test that the zip archive contains **all** the file names you expect, not only one.